doc/api/modules_pkg_config.md

```diff
@@ -0,0 +1,716 @@
+# Modules: Package configuration
+
+<!-- type=misc -->
+
+## Introduction
+
+A folder containing a `package.json` file, and all subfolders below that folder
+until the next folder containing another `package.json`, are a
+_package scope_.
+
+## Packages
+
+### Package entry points
```

Comment:
Before entry points we should include the info about `"type": "module"` and some basic modules v CommonJS questions. I believe moving that section from the ESM docs verbatim should cover it, but it may also need some massaging to make it more general.

Reply:
I'd like to do that in another PR if that's OK with you. My plan is actually to relive #32970 to document all the `package.json` fields.

Reply:
Agreed this involves more involved refactoring of the content that can be done as a follow-up. As long as that follow-up is on the roadmap then sure.